misc/services/mobile-webui/mobile-webui-frontend/src/containers/activities/common/LineScreen.jsx

```diff
@@ -39,14 +39,15 @@ class LineScreen extends PureComponent {
 }
 
 const mapStateToProps = (state, ownProps) => {
-  const { workflowId: wfProcessId, activityId, lineId } = ownProps.match.params;
+  const { workflowId: wfProcessId, activityId, lineId, appId } = ownProps.match.params;
   const wfProcess = selectWFProcessFromState(state, wfProcessId);
   const activity = wfProcess && wfProcess.activities ? wfProcess.activities[activityId] : null;
 
   const lineProps = activity != null ? activity.dataStored.lines[lineId] : null;
   const stepsById = lineProps != null && lineProps.steps ? lineProps.steps : {};
 
-  const appId = state.applications.activeApplication ? state.applications.activeApplication.id : null;
+  // for case when we load the app and routes, `appId` is not given yet
+  const stateAppId = state.applications.activeApplication ? state.applications.activeApplication.id : null;
```

Comment:
fine for now, if it works.....

but if we need to do more workarounds like this one,
then i would suggest introducing applicationId in URLs.

Reply:
Yeah, this was not really working as expected. Just used separate urls for everything